File: jobs-app-springboot/src/main/java/com/savaiz/jobs_app/entity/Job.java
package com.savaiz.jobs_app.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "jobs")
@Data
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "company")
    private String company;

    @Column(name = "location")
    private String location;

    @Column(name = "keywords")
    private List<String> keywords;

    @Column(name = "date_posted")
    private LocalDate date_posted;

    @Column(name = "created_at")
    private LocalDate created_at;

    @Column(name = "updated_at")
    private LocalDate updated_at;
}

File: jobs-app-python/models/job.py
from datetime import datetime


class Job:
    def __init__(self, title, description, company, location, keywords, date_posted, created_at=None, updated_at=None):
        self.title = title
        self.description = description
        self.company = company
        self.location = location
        self.keywords = keywords
        self.date_posted = date_posted
        self.created_at = created_at or datetime.now()
        self.updated_at = updated_at or datetime.now()

    def to_tuple(self):
        return (
            self.title,
            self.description,
            self.company,
            self.location,
            self.keywords,
            self.date_posted,
            self.created_at,
            self.updated_at
        )
